### ISSUE TITLE
False Positive marking of a Pydantic class as Dataclass

### ISSUE THREAD

**Issue by harsh-at-ikigailabs** [microsoft/pyright#11140]
## Environment data

-   Pylance version: 2025.8.3
-   OS and version: Mac OS 15.5 (24F74)
-   Python version (& distribution if applicable, e.g. Anaconda): 3.13

## Code Snippet

```python
from pydantic import BaseModel

class Flow(BaseModel):
    flow_id: str
    name: str
    __client: Client  # <-- Dataclass field cannot use private name

```

You can also clone: https://github.com/ikigailabs-io/ikigai and navigate to `src/ikigai/components/flow` around Line 177

## Expected behavior

The field `__client` should be treated like a normal field on the class as opposed to a field of a dataclass.

## Actual behavior

<img width="1423" height="239" alt="Image" src="https://github.com/user-attachments/assets/d2cfb791-c4be-4072-90d7-a996f1fe8fef" />

## Logs

[output.log](https://github.com/user-attachments/files/23064553/output.log)

**Comment by erictraut** [comment]
Pyright has no knowledge of Pydantic or its internal behaviors or limitations. It bases its analysis on type information provided to it and the rules spelled out in the Python typing standard.

Pydantic uses a feature of the Python type system called [dataclass_transform](https://typing.python.org/en/latest/spec/dataclasses.html#the-dataclass-transform-decorator). When a library like Pydantic uses this feature, it effectively tells a type checker "this class has dataclass-like behaviors". In this particular case, Pydantic's behavior deviates from that of standard dataclasses. 

So pyright is correctly implementing the typing spec here. This isn't a bug.
